lock-impl/src/test/java/com/palantir/lock/impl/AdjustableBackgroundTaskTest.java

```diff
@@ -136,6 +138,18 @@ public void suppliersAreCalledTogetherWithTask() {
         assertThat(numCallsToIntervalSupplier.get()).isEqualTo(11);
     }
 
+    @Test
+    public void closeShutsDownExecutorAndHaltsExecution() throws InterruptedException {
```

Comment:
- This fails if you switch the implementation to `shutdown` instead of `shutdownNow`
- The 30 seconds timeout is mostly for GC/scheduling. 30 seconds would be totally not OK in production (except for GC/scheduling ofc). This is why the delay used is very big, this means if the test passes we don't await the delayed task.